kernel/tlb_shootdown/src/lib.rs

```diff
@@ -50,26 +39,32 @@ pub fn handle_tlb_shootdown_ipi(pages_to_invalidate: PageRange) {
     // log::trace!("handle_tlb_shootdown_ipi(): AP {}, pages: {:?}", apic::current_cpu(), pages_to_invalidate);
 
     for page in pages_to_invalidate {
-        x86_64::instructions::tlb::flush(x86_64::VirtAddr::new(page.start_address().value() as u64));
+        tlb_flush_virt_addr(page.start_address());
     }
+
     TLB_SHOOTDOWN_IPI_COUNT.fetch_sub(1, Ordering::SeqCst);
 }
 
 
+/// Broadcasts TLB shootdown IPI to all other AP cores.
+///
+/// Do not invoke this directly, but rather pass it as a callback to the memory subsystem,
+/// which will invoke it as needed (on remap/unmap operations).
+///
 /// Sends an IPI to all other cores (except me) to trigger 
 /// a TLB flush of the given pages' virtual addresses.
-pub fn send_tlb_shootdown_ipi(my_lapic: &mut LocalApic, pages_to_invalidate: PageRange) {        
+fn broadcast_tlb_shootdown(pages_to_invalidate: PageRange) {        
     // skip sending IPIs if there are no other cores running
     let cpu_count = cpu_count();
     if cpu_count <= 1 {
         return;
     }
 
-    // log::trace!("send_tlb_shootdown_ipi(): from AP {}, cpu_count: {}, {:?}", my_lapic.apic_id(), cpu_count, pages_to_invalidate);
+    // log::trace!("send_tlb_shootdown_ipi(): cpu_count: {}, {:?}", cpu_count, pages_to_invalidate);
```

Comment:
let's keep the "from CPU" info too
```suggestion
    // log::trace!("send_tlb_shootdown_ipi(): from CPU {:?}, cpu_count: {}, {:?}", cpu::current_cpu(), cpu_count, pages_to_invalidate);
```

Reply:
Yep. I also changed this to the `if false` pattern.

Reply:
perfect!